From d07c0c549a8e825f985e7844644732dea91f9bf2 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Tue, 25 Aug 2020 11:10:31 +0200
Subject: [PATCH] nitrogen8mm: Add BalenaOS configs

Enable necessary configs for loading and updating
resinOS_uEnv.txt

Also, increase default env size to accomodate
for balenaOS scripts.

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 configs/nitrogen8mm_2g_defconfig | 4 ++++
 include/configs/nitrogen8mm.h    | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/configs/nitrogen8mm_2g_defconfig b/configs/nitrogen8mm_2g_defconfig
index 61ce65155e..4b8cf1e4e6 100644
--- a/configs/nitrogen8mm_2g_defconfig
+++ b/configs/nitrogen8mm_2g_defconfig
@@ -108,3 +108,7 @@ CONFIG_USBNET_DEVADDR="00:19:b8:00:00:02"
 CONFIG_USBNET_HOST_ADDR="00:19:b8:00:00:01"
 CONFIG_USB_HOST_ETHER=y
 CONFIG_LIBAVB=y
+CONFIG_ENV_IS_NOWHERE=y
+CONFIG_CMD_SAVEENV=n
+CONFIG_FAT_WRITE=y
+CONFIG_CMD_FS_UUID=y
diff --git a/include/configs/nitrogen8mm.h b/include/configs/nitrogen8mm.h
index 02eff617cd..a5e501be99 100644
--- a/include/configs/nitrogen8mm.h
+++ b/include/configs/nitrogen8mm.h
@@ -100,7 +100,7 @@
         (CONFIG_SYS_INIT_RAM_ADDR + CONFIG_SYS_INIT_SP_OFFSET)
 
 #define CONFIG_ENV_OVERWRITE
-#define CONFIG_ENV_SIZE			0x2000
+#define CONFIG_ENV_SIZE			0x3000
 #if defined(CONFIG_ENV_IS_IN_MMC)
 #define CONFIG_ENV_OFFSET               (-CONFIG_ENV_SIZE)
 #define CONFIG_SYS_MMC_ENV_DEV		0	/* USDHC1 */
-- 
2.17.1

